### PR TITLE
Don't error when calling `resolveEntityName`

### LINKED ISSUE #53247: Instantiation Expressions from generic class static member
# Bug Report

### 🔎 Search Terms

- "Only refers to a type, but is being used as a namespace here"  TS202
- "Cannot find name" TS2304
- "Instantiation Expressions"

### 🕗 Version & Regression Information

I tried this in Nightly 5.1.0-dev.20230313, 4.7.4, 4.8.4, and 4.9.5 - the versions that support [Instantiation Expressions](https://github.com/microsoft/TypeScript/pull/47607) on the playground but it does not complain.  This does not cause my build to fail - only the Language Server complains locally.

- This is the behavior in every version I tried

### ⏯ Playground Link

NOT exhibited in playground: [Playground link with relevant code](https://www.typescriptlang.org/play?esModuleInterop=false&target=99&ts=5.1.0-dev.20230313#code/MYGwhgzhAEBiD28A8AVaBTAHgF3QOwBMYB5AIwCt1hsAaaAYQx3yOgG8BfAPnYChoB0AA4AnAJYA3MLmjB4eCNhEBXavBEAKKSGXpoALmgo6cvLhyH6ASjbQOvfoMXSxwaCPRgC8kAE9oALZgANboCPDQALzQSIxYuIQwnFwapubYBgxWUTyoTAmsZJTUKdq6mSjZkTx46ADucIiodPSlYDroJvLpVgDcvPa8porQmFGN8AB0QaHhSMn9vJganH1AA)

But the same code in VSCode latest with nightly typescript:  

image

And I hope I'm showing that the right version of tsserver is being used (it would be nice if the TSServer startup announced its version for confirmation).

### 💻 Code

```ts
class Foo<T extends Object, C extends {}> {
    private constructor(value : T, context: C){ }

    static readonly makeFoo = <C extends {}>(context : C) => 
        <T extends Object>(value : T) => 
            new Foo<T, C>(value, context);
}

const x = Foo.makeFoo<{}>;  // Cannot find name 'Foo'.ts(2304)
                            // 'Foo' only refers to a type, but 
                            // is being used as a namespace here.ts(2702)

x({}); // 👍 `const x: <{}>(context: {}) => <T>(value: T) => Foo<T, {}>`
```

Strangely, the language server is still able to give the right types for that and future related code.

### 🙁 Actual behavior

Non-blocking errors TS202 TS2304 on the Instantiation Expressions involving a generic static method.

### 🙂 Expected behavior

Should be OK or have a fully blocking error (hopefully with enough information to fix)

### PR DESCRIPTION
**Pull request by gabritto**
Fixes #53247.

### CODE REVIEW

**Comment by gabritto** [comment]
@typescript-bot user test this
@typescript-bot run DT
@typescript-bot test top100

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -45308,7 +45308,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
 
             meaning |= SymbolFlags.Alias;
-            const entityNameSymbol = isEntityNameExpression(name) ? resolveEntityName(name, meaning) : undefined;
+            const entityNameSymbol = isEntityNameExpression(name) ? resolveEntityName(name, meaning, /*ignoreErrors*/ true) : undefined;
```
Comment (gabritto) [inline src/compiler/checker.ts:45311]: It's my understanding that we expect that this `resolveEntityName` call might fail, judging from the if statement below and the rest of the code (and an earlier resolve call that also ignores errors)`, so we shouldn't report errors here.

File: tests/cases/fourslash/classSymbolLookup.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+////     static readonly makeFoo = <C extends {}>(context : C) => 
+////         <T extends Object>(value : T) => 
+////             new Foo<T, C>(value, context);
+//// }
+
+//// const x = Foo.makeFoo<{}>;  //// Cannot find name 'Foo'.ts(2304)
+////                             //// 'Foo' only refers to a type, but 
+////                             //// is being used as a namespace here.ts(2702)
+
+//// x({}); // 👍 `const x: <{}>(context: {}) => <T>(value: T) => Foo<T, {}>`
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/classSymbolLookup.ts:17]: I think the comments are helpful for the issue, but are confusing here for the test case.

**Comment by DanielRosenwasser** [comment]
`getSymbolAtLocation` is *supposed* to be language servicey. Unfortunately

1. we use it elsewhere in the checker, so some real-world tests might catch something and are worth running.
2. we don't have real world language service consistency tests. :(

least we can do is test for stability/speed in the language service.

@typescript-bot user test tsserver
@typescript-bot test tsserver top100

**Pull request merged**